src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy

```diff
@@ -1183,6 +1198,47 @@ class MirrorEndpointsSpec extends HealthCheckSpecification {
         return northbound.getSwitchProperties(switchId)
     }
 
+    /**
+     * Identify all unique switches that need to be tested. Return minimum amount of switch pairs, where switch pair
+     * satisfies 'additional conditions' and src switch is a switch under test. Sources with 2 traffgens have a higher priority
+     */
+    @Memoized
+    List<SwitchPair> getUniqueSwitchPairs(Closure additionalConditions = { true }) {
+        def unpickedUniqueTgSwitches = topology.switches.findAll { it.traffGens }
```

Comment:
I guess you need to find all `active` traffgens.
I mean that traffGens can be `inactive` in the `topology.yamI` file, then as a result you will pick this traffgen, but won't be able to use them.

Reply:
fixed the 'getTraffgens' extension method to only return active traffgens